cmd/krel/cmd/promote.go

```diff
@@ -0,0 +1,254 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/release/pkg/command"
+	"k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/github"
+	"k8s.io/release/pkg/util"
+)
+
+const (
+	k8sioRepo             = "k8s.io"
+	k8sioManifestsPath    = "k8s.gcr.io"
+	stagingRepo           = "k8s-staging-kubernetes"
+	promotionBranchSuffix = "-image-promotion"
+)
+
+// promoteCommand is the krel subcommand to promote conainer images
+var promoteCommand = &cobra.Command{
+	Use:   "promote",
+	Short: "Starts an image promotion for a tag of kubernetes images",
+	Long: `krel promote
+
+The 'promote' subcommand of krel updates the image promoter manifests
+ans creates a PR in kubernetes/k8s.io`,
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Run the PR creation function
+		return runPromote()
```

Comment:
I suggest that we pass the `promoteOptions` in here to support testing.

Reply:
Done :+1: